### PR TITLE
add NatGatewayAvailable waiter

### PR DESCRIPTION
**Pull request by dpiddy**
Based `delay` and `maxAttempts` on `InstanceRunning`, happy to change if there are better values to use.

### CODE REVIEW

**Comment by dpiddy** [comment]
@awood45 anything I can do to help this along?

**Comment by awood45** [comment]
I'm testing this now.

File: aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json [hunk 1]
```diff
@@ -281,6 +281,42 @@
...
     },
+    "NatGatewayAvailable": {
+      "operation": "DescribeNatGateways",
+      "delay": 15,
+      "maxAttempts": 40,
+      "acceptors": [
+        {
+          "state": "success",
+          "matcher": "pathAll",
+          "argument": "NatGateways[].Status",
```
Comment (awood45) [inline aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json]: s/Status/State

File: aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json [hunk 2]
```diff
@@ -281,6 +281,42 @@
...
+          "expected": "available"
+        },
+        {
+          "state": "failure",
+          "matcher": "pathAny",
+          "argument": "NatGateways[].Status",
```
Comment (awood45) [inline aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json]: s/Status/State

File: aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json [hunk 3]
```diff
@@ -281,6 +281,42 @@
...
+          "expected": "failed"
+        },
+        {
+          "state": "failure",
+          "matcher": "pathAny",
+          "argument": "NatGateways[].Status",
```
Comment (awood45) [inline aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json]: s/Status/State

File: aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json [hunk 4]
```diff
@@ -281,6 +281,42 @@
...
+          "expected": "deleting"
+        },
+        {
+          "state": "failure",
+          "matcher": "pathAny",
+          "argument": "NatGateways[].Status",
```
Comment (awood45) [inline aws-sdk-core/apis/ec2/2015-10-01/waiters-2.json]: s/Status/State

**Comment by awood45** [comment]
Looks good once the incorrect matcher is fixed.

**Comment by dpiddy** [comment]
Fixed!

**Comment by awood45** [comment]
Thanks!

**Pull request merged**